edgelet/edgelet-http-workload/src/server/decrypt.rs

```diff
@@ -1,38 +1,43 @@
 // Copyright (c) Microsoft. All rights reserved.
+use std::sync::{Arc, Mutex};
 
 use failure::ResultExt;
 use futures::{future, Future, IntoFuture, Stream};
 use hyper::header::{CONTENT_LENGTH, CONTENT_TYPE};
 use hyper::{Body, Request, Response, StatusCode};
 
-use edgelet_core::Decrypt;
+use aziot_key_common::EncryptMechanism;
+use aziot_key_common::KeyHandle;
 use edgelet_http::route::{Handler, Parameters};
 use edgelet_http::Error as HttpError;
+use identity_client::client::IdentityClient;
 use workload::models::{DecryptRequest, DecryptResponse};
 
+use super::get_key_handle;
+
 use crate::error::{EncryptionOperation, Error, ErrorKind};
 use crate::IntoResponse;
 
-pub struct DecryptHandler<T: Decrypt> {
-    hsm: T,
+pub struct DecryptHandler {
+    key_client: Arc<Mutex<aziot_key_client::Client>>,
+    identity_client: Arc<Mutex<IdentityClient>>,
 }
 
-impl<T: Decrypt> DecryptHandler<T> {
-    pub fn new(hsm: T) -> Self {
-        DecryptHandler { hsm }
+impl DecryptHandler {
+    pub fn new(key_client: Arc<Mutex<aziot_key_client::Client>>, identity_client: Arc<Mutex<IdentityClient>>) -> Self {
```

Comment:
You don't need a `Mutex` around `aziot_key_client::Client`. All its functions take `&self`

Reply:
Fixed